src/t-i18n.ts

```diff
@@ -1,132 +1,103 @@
 import { IcuReplacements, Messages, MFunc, SetupOptions, AnyReplacements } from "./types";
-import createCachedFormatter, { CachedFormatter, numberFormatOptions, dateTimeFormatOptions} from "./format";
+import createCachedFormatter, { numberFormats, dateTimeFormats} from "./format";
 import { generator, assign, splitAndEscapeReplacements } from "./helpers";
 import parseIcu from "./icu";
 import parseXml from "./xml";
 
 /**
  * T-i18n - lightweight localization
- * v0.5.0
+ * v0.6.0
  *
  * T-i18n defers to standards to do the hard work of localization. The browser Intl API is use to format
  * dates and numbers. Messages are provided as functions rather than strings, so they can be compiled at build time.
- *
- *
- * Cobbled together by Mitch Cohen on July 31, 2017
  */
 
-// T is exported instead of the underlying instance to allow calls to T("Message to translate")
-//
 export interface TFunc {
 	(message: string, replacements?: IcuReplacements, id?: string): string;
+	generateId: (message: string) => string;
+	locale: () => string;
 	lookup: (id: string, replacements?: IcuReplacements, defaultMessage?:string) => string
 	setup: (options?: SetupOptions) => any;
-	date: (value: any, formatName?: string, locale?: string) => string;
-	number: (value: any, formatName?: string, locale?: string) => string;
+	date: (value: any, formatName?: keyof typeof dateTimeFormats, locale?: string) => string;
```

Comment:
Unless I'm mistaken, the `value` for both the date and number functions is `number`. They should be typed regardless.

BTW, it's outside the scope of this PR but I'd love to be able to pass in custom formatter options.